### PR TITLE
Test all cases for LIST request cache bypass

### PR DESCRIPTION
**Pull request by serathius**
/kind cleanup

```release-note
NONE
```

Extracted from https://github.com/kubernetes/kubernetes/pull/118508
/assign @wojtek-t

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go [hunk 1]
```diff
@@ -154,41 +154,52 @@ func (d *dummyStorage) injectError(err error) {
...
-	}
-
-	err = cacher.GetList(context.TODO(), "pods/ns", storage.ListOptions{
-		ResourceVersion: "",
-		Predicate:       pred,
-		Recursive:       true,
-	}, result)
-	if err != errDummy {
-		t.Errorf("GetList with Limit without RV=0 should bypass cacher: %v", err)
+	gotBypass := cacher.GetList(context.TODO(), "pods/ns", options, result) == errDummy
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go:200]: Let's be more specific here maybe.
```
err := cacher.GetList(...)
if expectBypass {
  if err != errDummy {
    t.Errrof(...)
   }
} else {
  if err != nil {
    t.Errorf(...)
  }
}
```

Currently if we got an error in non-bypass case we won't catch it.

Comment (serathius) [reply staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go:200]: Right

**Pull request merged**